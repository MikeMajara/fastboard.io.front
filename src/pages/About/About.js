import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { Button, HeadMeta } from '#components';
import routes from '#routes';
import { analyticsService } from '#services';
import collaborateImg from '#theme/images/about-collaborate.jpg';
import easyImg from '#theme/images/about-easy.jpg';
import openImg from '#theme/images/about-open.jpg';

import {
  SWrapper,
  SHeader,
  SSubHeader,
  SContent,
  SSection,
  SSectionImage,
  SSectionText,
  SSectionTextHeader,
} from './styled';

const About = () => {
  const { t } = useTranslation('about');
  const { push : redirectTo } = useHistory();

  // Analytics pageview
  useEffect(() => analyticsService.pageview(routes.about), []);

  const onStart = () => redirectTo(routes.home);

  return (
    <>
      <HeadMeta route={routes.about} />
      <SWrapper>
        <SHeader>
          {t('title')}
        </SHeader>
        <SSubHeader>
          {t('main.t1')}
        </SSubHeader>
        <SSubHeader>
          {t('main.t2')}
        </SSubHeader>
        <SContent>
          {t('main.t3')}
        </SContent>
        <SContent extraSpace>
          <Button size="md" onClick={onStart}>
            {t('start')}
          </Button>
        </SContent>
        <SContent extraSpace>
          <SSection>
            <SSectionImage image={collaborateImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('collaborate.header')}
              </SSectionTextHeader>
              <div>
                {t('collaborate.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={easyImg} invert />
            <SSectionText>
              <SSectionTextHeader>
                {t('easy.header')}
              </SSectionTextHeader>
              <div>
                {t('easy.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={openImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('open.header')}
              </SSectionTextHeader>
              <div>
                {t('open.content')}
              </div>
            </SSectionText>
          </SSection>
        </SContent>
      </SWrapper>
    </>
  );
};

export default About;
