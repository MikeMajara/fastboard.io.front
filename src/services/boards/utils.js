/**
 * Send boards event trough the network.
 *
 * @param {String} eventName Event key name.
 * @param {Object} data Data associated to the event.
 * @param {Function} ask Callback for server event acknowledgement.
 */
function send(eventName, data, ack) {
  this.socket.emit(eventName, data, ack);
}

/**
 * Check if the service is initialized.
 *
 * @return {Boolean} True if the service is initialized; false otherwise.
 */
function isInit() {
  return this.isInit;
}

export {
  isInit,
  send,
};
