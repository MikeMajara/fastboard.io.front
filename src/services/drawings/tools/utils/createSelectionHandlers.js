import { Group, Shape } from 'paper';

import { Log } from '#utils';
import { cursorTypes, selectionColorCode } from '#constants';
import store, { setSelectorCursorHover } from '#store';

import { bounds } from '../selector/constants';

const boundsArray = Object.values(bounds);
const bounds2cursorType = Object.freeze({
  [bounds.topLeft]     : cursorTypes.rotating,
  [bounds.topRight]    : cursorTypes.menu,
  [bounds.bottomLeft]  : cursorTypes.resizingSW,
  [bounds.bottomRight] : cursorTypes.resizingSE,
});

/**
 * Add selection handlers to the selected item.
 *
 * @param {Object} selectedItem Item to be added selection handlers to.
 * @param {Object} selectionLayer Layer that will contain the selection handlers.
 *
 * @returns {Object} Item - group with 4 handler items - from the selection handler layer.
 */
export default function createSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : createSelectionHandlers', { selectedItem });

  // Set selected item to show its selection state
  selectedItem.set({
    selected      : true,
    selectedColor : selectionColorCode,
    onMouseEnter  : () => store.dispatch(setSelectorCursorHover(cursorTypes.dragging)),
    onMouseLeave  : () => store.dispatch(setSelectorCursorHover()),
  });

  // Create bound handlers on the selection layer
  boundsArray.forEach((bound) => {
    selectionLayer.addChild(new Shape.Circle({
      name         : bound,
      center       : selectedItem.internalBounds[bound],
      radius       : 6,
      fillColor    : selectionColorCode,
      onMouseEnter : () => store.dispatch(setSelectorCursorHover(bounds2cursorType[bound])),
      onMouseLeave : () => store.dispatch(setSelectorCursorHover()),
    }));
  });

  // Create a group with the handlers to manipulate all of them at once
  selectionLayer.addChild(
    new Group({
      name     : 'selectionHandlers', // @todo to constants
      children : boundsArray.map(
        (bound) => selectionLayer.children[bound],
      ),
    }),
  );

  // Reproduce the selected item transformations on the handler items group
  selectionLayer.children.selectionHandlers.transform(selectedItem.matrix);

  // Except the scaling of the handler items themselves, which retain the original size
  selectionLayer.children.selectionHandlers.children.forEach((handlerItem) => {
    handlerItem.set({ scaling : 1 });
  });

  return selectionLayer.children.selectionHandlers;
}
