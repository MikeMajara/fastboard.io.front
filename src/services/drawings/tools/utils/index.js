export { default as createSelectionHandlers } from './createSelectionHandlers';
export { default as removeSelectionHandlers } from './removeSelectionHandlers';
export { default as resizeSelectionHandlers } from './resizeSelectionHandlers';
export { default as rotateSelectionHandlers } from './rotateSelectionHandlers';
export { default as translateSelectionHandlers } from './translateSelectionHandlers';
