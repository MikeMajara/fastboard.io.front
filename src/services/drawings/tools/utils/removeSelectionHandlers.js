import { Log } from '#utils';

/**
 * Remove selection handlers.
 *
 * @param {Object} selectedItem Item to be removed selection handlers from.
 * @param {Object} selectionLayer Layer that contains the selection handlers.
 */
export default function removeSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : removeSelectionHandlers');

  selectedItem.set({ selected : false });
  selectionLayer.removeChildren();
}
