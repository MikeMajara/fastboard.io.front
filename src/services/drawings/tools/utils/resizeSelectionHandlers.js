import { Log } from '#utils';

/**
 * @todo
 * @param {} selectedItem
 * @param {*} handlersItem
 */
export default function resizeSelectionHandlers(selectedItem, handlersItem) {
  Log.debug('Services : Drawings : Tools : Utils : resizeSelectionHandlers');

  handlersItem.children.topLeft.position.set(selectedItem.bounds.topLeft);
  handlersItem.children.topRight.position.set(selectedItem.bounds.topRight);
  handlersItem.children.bottomLeft.position.set(selectedItem.bounds.bottomLeft);
  handlersItem.children.bottomRight.position.set(selectedItem.bounds.bottomRight);
}
