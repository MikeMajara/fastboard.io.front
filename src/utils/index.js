export { default as Log } from './logger';
export { default as generateId } from './generateId';
export { default as point2net } from './point2net';
export { default as setPreventUnload } from './setPreventUnload';
export { default as throttle } from './throttle';
export { default as timeoutPromise } from './timeoutPromise';
export { default as triggerDownload } from './triggerDownload';
export { default as triggerUpload } from './triggerUpload';
export { default as validBoardId } from './validBoardId';
