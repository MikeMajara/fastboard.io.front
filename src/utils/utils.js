/**
 * Throttles a function call.
 *
 * @param {Function} callback Function to be throttled.
 * @param {Integer} delay Throttle delay in milliseconds.
 *
 * @return {Function} Function that throttles the execution of the callback.
 */
export const throttle = (callback, delay = 0) => {
  let previousTime = new Date().getTime();

  return function throttledCallback(args) {
    const currentTime = new Date().getTime();

    if ((currentTime - previousTime) >= delay) {
      previousTime = currentTime;
      callback(...args);
    }
  };
};

/**
 * Creates a Promise with a rejection timeout.
 *
 * @param {Function} callback Function for the promise. Receives res and rej functions.
 * @param {Integer} timeout Timeout for the promise to be rejected in milliseconds.
 *
 * @return {Promise} Promise with rejection timeout.
 */
export const timeoutPromise = (callback, timeout) => new Promise(
  (res, rej) => {
    const timeoutId = setTimeout(() => rej(new Error()), timeout);

    callback(
      (...args) => {
        clearTimeout(timeoutId);
        res(...args);
      },
      (...args) => {
        clearTimeout(timeoutId);
        rej(new Error(...args));
      },
    );
  },
);
