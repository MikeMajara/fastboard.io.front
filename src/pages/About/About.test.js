import React from 'react';
import { HelmetProvider } from 'react-helmet-async';

import { render } from '#devTools/tests/utils';

import About from './About';

jest.mock('#services', () => ({
  analyticsService : {
    pageview : jest.fn(),
  },
}));

describe('Page : About', () => {
  test('Snapshot', () => {
    const { asFragment } = render(
      <HelmetProvider>
        <About />
      </HelmetProvider>,
      { router : true },
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
