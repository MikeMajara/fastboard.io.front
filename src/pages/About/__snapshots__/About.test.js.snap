// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page : About Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-prQdK dobkOb"
  >
    <h1
      class="sc-qQlgh dEyskS"
    >
      title
    </h1>
    <h2
      class="sc-qYhTA jRQCrj"
    >
      main.t1
    </h2>
    <h2
      class="sc-qYhTA jRQCrj"
    >
      main.t2
    </h2>
    <div
      class="sc-pAYXY ciGxoL"
    >
      main.t3
    </div>
    <div
      class="sc-pAYXY hgjvpw"
    >
      <button
        class="sc-qQYBZ hDUFJj"
        type="primary"
      >
        start
      </button>
    </div>
    <div
      class="sc-pAYXY hgjvpw"
    >
      <div
        class="sc-pJwpB cuHQYa"
      >
        <div
          class="sc-pRSgm kPrgJd"
        />
        <div
          class="sc-qapxP gLxKKc"
        >
          <h3
            class="sc-oTmHo dhXOX"
          >
            collaborate.header
          </h3>
          <div>
            collaborate.content
          </div>
        </div>
      </div>
      <div
        class="sc-pJwpB cuHQYa"
      >
        <div
          class="sc-pRSgm goWzTJ"
        />
        <div
          class="sc-qapxP gLxKKc"
        >
          <h3
            class="sc-oTmHo dhXOX"
          >
            easy.header
          </h3>
          <div>
            easy.content
          </div>
        </div>
      </div>
      <div
        class="sc-pJwpB cuHQYa"
      >
        <div
          class="sc-pRSgm kPrgJd"
        />
        <div
          class="sc-qapxP gLxKKc"
        >
          <h3
            class="sc-oTmHo dhXOX"
          >
            open.header
          </h3>
          <div>
            open.content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
